lib/core/versions/latest/CreateOperation.ts

```diff
@@ -99,53 +99,70 @@ export default class CreateOperation implements OperationModel {
+   * NOTE: This method is purely intended to be used as an optimization method over the `parse` method in that
+   * JSON parsing is not required to be performed more than once when an operation buffer of an unknown operation type is given.
    * @param operationObject The operationObject is a json object with no encoding
    * @param operationBuffer The buffer format of the operationObject
+   * @param anchorFileMode If set to true, then `delta` and `type` properties are expected to be absent.
    */
-  public static parseJcsObject (operationObject: any, operationBuffer: Buffer): CreateOperation {
+  public static parseJcsObject (operationObject: any, operationBuffer: Buffer, anchorFileMode: boolean): CreateOperation {
     let expectedPropertyCount = 3;
+    if (anchorFileMode) {
+      expectedPropertyCount = 1;
+    }
     const properties = Object.keys(operationObject);
     if (properties.length !== expectedPropertyCount) {
       throw new SidetreeError(ErrorCode.CreateOperationMissingOrUnknownProperty);
     }
 
-    if (operationObject.type !== OperationType.Create) {
-      throw new SidetreeError(ErrorCode.CreateOperationTypeIncorrect);
-    }
-
     CreateOperation.validateSuffixData(operationObject.suffix_data);
-    const suffixData = {
+    const suffixData: SuffixDataModel = {
       deltaHash: operationObject.suffix_data.delta_hash,
       recoveryCommitment: operationObject.suffix_data.recovery_commitment
     };
 
+    if (operationObject.suffix_data.type !== undefined) {
+      suffixData.type = operationObject.suffix_data.type;
+    }
+
     // For compatibility with data pruning, we have to assume that `delta` may be unavailable,
     // thus an operation with invalid `delta` needs to be processed as an operation with unavailable `delta`,
     // so here we let `delta` be `undefined`.
     let delta;
-    try {
-      Operation.validateDelta(operationObject.delta);
-      delta = {
-        patches: operationObject.delta.patches,
-        updateCommitment: operationObject.delta.update_commitment
-      };
-    } catch {
-      delta = undefined;
+    let encodedDelta;
+    if (!anchorFileMode) {
+      if (operationObject.type !== OperationType.Create) {
+        throw new SidetreeError(ErrorCode.CreateOperationTypeIncorrect);
+      }
+
+      try {
+        Operation.validateDelta(operationObject.delta);
+        delta = {
+          patches: operationObject.delta.patches,
+          updateCommitment: operationObject.delta.update_commitment
+        };
+      } catch {
+        // For compatibility with data pruning, we have to assume that `delta` may be unavailable,
+        // thus an operation with invalid `delta` needs to be processed as an operation with unavailable `delta`,
+        // so here we let `delta` be `undefined`.
+      }
+      encodedDelta = Encoder.encode(JsonCanonicalizer.canonicalizeAsBuffer(operationObject.delta));
```

Comment:
Sanity: I suppose this is temporary right? As in this encoded string is not going to be useable for the other code path, the property is probably entirely unnecessary after SIP2.

Reply:
Yes, this is just here because createOperation still needs it for now.